src/iam.ts

```diff
@@ -125,15 +126,24 @@ export class IAM extends IAMBase {
   async initializeConnection(
     {
       useMetamaskExtension,
-      reinitializeMetamask
-    }: { useMetamaskExtension: boolean; reinitializeMetamask?: boolean } = {
-      useMetamaskExtension: false
+      reinitializeMetamask,
+      walletProvider
+    }: { useMetamaskExtension?: boolean; reinitializeMetamask?: boolean, walletProvider?: WalletProvider } = {
     }
   ): Promise<InitializeData> {
     try {
+      // Deprecated warning below can be removed after param is removed
+      if (useMetamaskExtension !== undefined) {
+        console.warn("Use of useMetamaskExtension parameter is deprecated. Use walletProvider enumeration instead.")
```

Comment:
@dwojno @manihagh Do you agree with this approach of signalling that a param is deprecated to users of `iam-client-lib`

Reply:
@jrhender I think we should just switch to new approach using walletProvider enum, as for now only switchboard is using iam-client-lib so it should be fine. I think leaving it both may cause a lot of mess in initializing connection methods on iam-client-lib, what do you think?

Reply:
@dwojno Thanks for the advice, I removed the `useMetamaskExtension` param